bin/install-runtime.js

```diff
@@ -0,0 +1,281 @@
+        }
+      });
+    }
+    else if (process.platform === 'linux') {
+      const source = filePath;
+      const destination = distDir;
+      return pify(fs.remove)(path.join(destination, 'runtime'))
+      .then(() => {
+        return decompress(source, destination);
+      })
+      .then(() => {
+        return pify(fs.rename)(path.join(destination, 'firefox'), path.join(destination, 'runtime'));
+      });
+    }
+  })
+  .then(() => {
+    return installXULApp();
+  })
+  .then(() => {
+    // Expand the browser xulapp's JAR archive so we can access its devtools.
+    // We have to expand it into a subdirectory of qbrt's xulapp directory,
+    // because chrome manifests can't reference super-directories.
+
+    // TODO: limit expansion to browser files that are necessary for devtools.
+
+    const targetDir = path.join(resourcesDir, 'qbrt', 'browser');
+
+    // "decompress" fails silently on omni.ja, so we use extract-zip here instead.
+    // TODO: figure out the issue with "decompress" (f.e. that the .ja file
+    // extension is unrecognized or that the chrome.manifest file in the archive
+    // conflicts with the one already on disk).
+    return pify(extract)(browserJAR, { dir: targetDir });
+  })
+  .then(() => {
+    // Copy devtools pref files from browser to qbrt.
+
+    const sourceDir = path.join(resourcesDir, 'qbrt', 'browser', 'defaults', 'preferences');
+    const targetDir = path.join(resourcesDir, 'qbrt', 'defaults', 'preferences');
+
+    const prefFiles = [
+      'debugger.js',
+      'devtools.js',
+    ];
+
+    return Promise.all(prefFiles.map(file => pify(fs.copy)(path.join(sourceDir, file), path.join(targetDir, file))));
+  })
+  .then(() => {
+    // Copy and configure the stub executable.
+
+    switch (process.platform) {
+      case 'win32': {
+        // Copy the stub executable to the executable dir.
+        return pify(fs.copy)(path.join(__dirname, '..', 'launcher.bat'), path.join(executableDir, 'launcher.bat'));
+      }
+      case 'darwin': {
+        return pify(fs.copy)(path.join(__dirname, '..', 'launcher.sh'), path.join(executableDir, 'launcher.sh'))
+        .then(() => {
+          // Configure the bundle to run the stub executable.
+          const plistFile = path.join(installDir, 'Contents', 'Info.plist');
+          return pify(plist.readFile)(plistFile)
+          .then(appPlist => {
+            appPlist.CFBundleExecutable = 'launcher.sh';
+            return pify(plist.writeFile)(plistFile, appPlist);
+          });
+        });
+      }
+      case 'linux': {
+        // Copy the stub executable to the executable dir.
+        return pify(fs.copy)(path.join(__dirname, '..', 'launcher.sh'), path.join(executableDir, 'launcher.sh'));
+      }
+    }
+  })
+  .catch(error => {
+    if (fileStream) {
+      fileStream.end();
+    }
+    throw error;
+  })
+  .finally(() => {
+    return pify(fs.remove)(tempDir);
+    // XXX Remove partial copy of Firefox.
```

Comment:
I ended up removing this comment (6d71957), since on second thought, if the Firefox install failed, it may still be useful to leave the partial install around until the user repeats the installation process.

I also removed another "XXX" comment (3f0cc81) that was just explaining something being done rather than suggesting that it should be done differently.

Those were the only two XXX comments in the project, so there shouldn't be any more, and 5ca9f7e adds no-warning-comments to .eslintrc.js to prevent additional ones from being added. However, note that it currently exempts TODO comments, since the codebase already contains a bunch of those, so we'll need to get rid of them before we can remove that exemption.